lib/engine.py

```diff
@@ -687,7 +687,7 @@ def to_csv(self):
             csv_file = open_fw(csv_file_output)
             csv_writer = open_csvw(csv_file)
             self.get_cursor()
-            self.set_engine_encoding()
+            self.set_engine_encoding(ENCODING)
```

Comment:
This would pass the default encoding value, which I think would force all engines to set the encoding to the default. I think we want to be passing the actual encoding value instead.